### PR TITLE
REF: type change `escape` in `Styler.format` to str to allow "html" and "latex"

### PR DESCRIPTION
**Pull request by attack68**
Change is made ahead of 1.3.0 to pre-empt the addition of `escape_latex` and to avoid name ambiguity.

This is not user facing since the `escape` kwarg was added for 1.3.0 only recently.

**Alternative**

If instead of having separate `escape_html` and `escape_latex` kwargs we could have `escape={None, 'html', 'latex'}` since it is likely they will be used exclusively??

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
can u move this to the to_html or to_latex method instead?

**Comment by attack68** [comment]
> can u move this to the to_html or to_latex method instead?

no, because it's mixed in as part of the formatting routine, and the idea is that DataFrames should be formatted before calling `to_html` or `to_latex`, i.e.

```df.style.format(...).to_xxx()```

if escape is added to the render method then `format` needs to be called within the render method overwriting the previous chained application and negating it all and restricting its flexibility.

what about expanding the keyword inputs to:

```
escape=None,
escape='html'
escape='latex'
escape=['html', 'latex']
```

?

**Comment by jreback** [comment]
yeah the last is fine i guess

but is there a way to just escape things that work in both html and latex?

**Comment by attack68** [comment]
> yeah the last is fine i guess
> 
> but is there a way to just escape things that work in both html and latex?

not currently because of operation order. the format method is ignorant of how the styler will ultimately be rendered, and it yields a one-arg-callable for each data element decoupled from render method. if it becomes dependent it introduces operation order complexity and possible performance degradation. 

will propose the former alternative and can review again.

**Comment by attack68** [comment]
@jreback it turns out some of the escaped characters over;ap so it is **impossible** to escape 'html' and 'latex' simultaneously.

I've edited this PR to allow `escape in {None, 'html', 'latex'}` and added the necessary latex escape function and tests.

File: pandas/io/formats/style_render.py [hunk 1]
```diff
@@ -771,10 +788,13 @@ def wrapper(x):
...
-def _str_escape_html(x):
-    """if escaping html: only use on str, else return input"""
+def _str_escape(x, escape):
+    """if escaping: only use on str, else return input"""
     if isinstance(x, str):
-        return escape_html(x)
+        if escape == "html":
+            return escape_html(x)
+        elif escape == "latex":
+            return _escape_latex(x)
```
Comment (jreback) [inline pandas/io/formats/style_render.py:797]: if it's anything else (not None) it's an error

pls add a test

File: pandas/io/formats/style_render.py [hunk 2]
```diff
@@ -1187,3 +1207,38 @@ def _parse_latex_options_strip(value: str | int | float, arg: str) -> str:
     For example: 'red /* --wrap */  ' --> 'red'
     """
     return str(value).replace(arg, "").replace("/*", "").replace("*/", "").strip()
+
+
+def _escape_latex(s):
```
Comment (jreback) [inline pandas/io/formats/style_render.py:1212]: where is eacape_html?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style_render.py (hunk 2 above)
Comment (attack68) [reply pandas/io/formats/style_render.py:1212]: imported from markupsafe which is a jinja2 dependency

File: pandas/io/formats/style_render.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style_render.py:797]: done

File: pandas/io/formats/style_render.py (hunk 2 above)
Comment (attack68) [reply pandas/io/formats/style_render.py:1216]: it is a very simple function, can write here instead of having dependency to markupsafe if preferred?

Comment (jreback) [reply pandas/io/formats/style_render.py:1216]: oh its fine, just not completely obvious *where* it is (but its already there so its fine)

**Review by jreback — APPROVED** [review]

**Pull request merged**